### PR TITLE
Remove no-op DrawBuffer::End(), assorted additional cleanup in the GPU backends

### PR DESCRIPTION
**Pull request by hrydgard**

### CODE REVIEW

**Comment by hrydgard** [comment]
Threw in some more cleanup.

**Comment by hrydgard** [comment]
The build errors are just the usual Travis connectivity nonsense.

File: GPU/Directx9/StencilBufferDX9.cpp [hunk 1]
```diff
@@ -33,36 +33,38 @@ namespace DX9 {
...
-"  float4 position : POSITION;\n"
-"  float2 v_texcoord0 : TEXCOORD0;\n"
-"};\n"
-"VS_OUT main(VS_IN In) {\n"
-"  VS_OUT Out;\n"
-"  Out.position = In.a_position;\n"
-"  Out.v_texcoord0 = In.a_texcoord0;\n"
-"  return Out;\n"
-"}\n";
+float4 u_stencilValue : register(c" STR(CONST_PS_STENCILVALUE) ");
```
Comment (unknownbrackets) [inline GPU/Directx9/StencilBufferDX9.cpp]: Needs to be `)" STR(CONST_PS_STENCILVALUE) R"(`.  This will just error.

-[Unknown]

Comment (hrydgard) [reply GPU/Directx9/StencilBufferDX9.cpp]: Oops, will push a fix.

**Pull request merged**